velox/common/memory/MemoryUsageTracker.h

```diff
@@ -201,7 +216,52 @@ class MemoryUsageTracker
   }
 
  private:
+  static constexpr int64_t kMinQuantum = 1 << 20;
+
   enum class UsageType : int { kUserMem = 0, kSystemMem = 1, kTotalMem = 2 };
+
+  int64_t reserveInternal(int64_t size) {
+    int64_t neededSize = size - (reservation_ - usedReservation_);
+    if (neededSize > 0) {
+      auto increment = roundedDelta(reservation_, neededSize);
+      updateInternal(type_, increment);
+      reservation_ += increment;
+      return increment;
+    }
+    return 0;
+  }
+
+  // Returns a rounded up delta based on adding 'delta' to
+  // 'size'. Adding the rounded delta to 'size' will result in 'size'
+  // a quantized size, rounded to the MB or 8MB for larger sizes.
+  int64_t roundedDelta(int64_t size, int64_t delta) {
+    return quantizedSize(size + delta) - size;
+  }
+
+  // Returns the next higher quantized size. Small sizes are at MB granularity,
+  // larger ones at coarser granularity.
+  uint64_t quantizedSize(uint64_t size) {
+    if (size < 16 << 20) {
```

Comment:
nit: this code might be easier to read if N << 20 expressions were replaced with constants: k16MB, k64MB, k1MB, etc.